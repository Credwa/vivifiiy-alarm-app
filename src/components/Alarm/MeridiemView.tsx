import React, { useEffect, useState } from 'react';
import ScrollSelector from '@/components/ScrollSelector';
import Colors from '@/constants/Colors';
import { meridiem } from '@/types';
import useStore from '@/store';
import { findSelectedAlarmViewIndex, resize } from '@/utils';
import { FontName } from '../VivText';
interface SnapScrollProps {
  data: meridiem[];
  initValue: meridiem;
}

export default function MeridiemView({ data, initValue }: SnapScrollProps) {
  const [listState, setEndOfList] = useState(data);
  const updateMeridiemOnChange = useStore((state) => state.setMeridiem);
  useEffect(() => {
    updateMeridiemOnChange(initValue);
  }, []);

  return (
    <ScrollSelector
      dataSource={listState}
      overrideFontName={resize<FontName[]>(['Title3', 'Title4'], ['Title5', 'Title6'], ['Title1', 'Title2'])}
      selectedIndex={findSelectedAlarmViewIndex(data, initValue, 0)}
      onValueChange={(data) => {
        updateMeridiemOnChange(data);
      }}
      wrapperHeight={resize<number>(200, 175, 300)}
      wrapperWidth={resize<number>(80, 55, 150)}
      itemHeight={resize<number>(75, 65, 110)}
      highlightColor={Colors.greyLight3}
      highlightBorderWidth={1}
    />
  );
}
