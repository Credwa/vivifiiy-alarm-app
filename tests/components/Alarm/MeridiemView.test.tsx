import React from 'react';
import renderer from 'react-test-renderer';
import MeridiemView from '@/components/Alarm/MeridiemView';

jest.useFakeTimers();
describe('<MeridiemView />', () => {
  it(`renders correctly`, () => {
    const tree = renderer.create(<MeridiemView data={['AM', 'PM']} initValue={'AM'} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
