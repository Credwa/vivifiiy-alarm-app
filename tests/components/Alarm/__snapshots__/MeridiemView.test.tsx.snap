// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MeridiemView /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignSelf": "center",
      "backgroundColor": null,
      "height": 200,
      "overflow": "hidden",
      "width": 80,
    }
  }
>
  <View
    style={
      Object {
        "borderBottomColor": "#B1C2D1",
        "borderBottomWidth": 1,
        "borderTopColor": "#B1C2D1",
        "borderTopWidth": 1,
        "height": 75,
        "position": "absolute",
        "top": 62.5,
        "width": "100%",
      }
    }
  />
  <RCTScrollView
    bounces={false}
    onMomentumScrollBegin={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    scrollEventThrottle={400}
    showsVerticalScrollIndicator={false}
  >
    <View>
      <View
        style={
          Object {
            "flex": 1,
            "height": 62.5,
          }
        }
      />
      <View
        style={
          Object {
            "alignItems": "center",
            "height": 75,
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#869DB2",
              },
              Object {
                "fontFamily": "Nunito_400Regular",
                "fontSize": 28,
              },
              Object {
                "color": "#869DB2",
              },
              undefined,
            ]
          }
        >
          AM
        </Text>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "height": 75,
            "justifyContent": "center",
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#869DB2",
              },
              Object {
                "fontFamily": "Nunito_400Regular",
                "fontSize": 34,
              },
              Object {
                "color": "#CEE1F2",
              },
              undefined,
            ]
          }
        >
          PM
        </Text>
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "height": 62.5,
          }
        }
      />
    </View>
  </RCTScrollView>
</View>
`;
